main.go

```diff
@@ -131,6 +132,7 @@ func main() {
 	flag.StringVar(&logLevel, "log-level", logging.LogLevelDebug, `The verbosity of the logging. Valid values are "debug", "info", "warn", "error". Defaults to "debug".`)
 	flag.StringVar(&logFormat, "log-format", "text", `The log format. Valid options are "text" and "json". Defaults to "text"`)
 	flag.BoolVar(&autoScalingRunnerSetOnly, "auto-scaling-runner-set-only", false, "Make controller only reconcile AutoRunnerScaleSet object.")
+	flag.BoolVar(&drainJobsMode, "drain-jobs-mode", false, "Wait for jobs to finish before mutating resources.")
```

Comment:
Suggestion:

Should we use `update-strategy` as a flag and assign strategies to it?
To turn on draining: `--update-strategy=drain`
To use eventual replace: `--update-strategy=replace` 
Or something similar. We can always have a default value if no strategy is specified

Reply:
```
  ## Defines how the controller should handle upgrades while having running jobs.
  ##
  ## The srategies available are:
  ## - "immediate": (default) The controller will immediately appy the change leading to the
  ##   recreation of the listener and ephemeral runner set. This can lead to an 
  ##   overprovisioning of runners if there are pending / running jobs. This should not
  ##   be a problem at a small scale but it could lead to a significant increase of 
  ##   resources if you have a lot of jobs running concurrently.
  ## 
  ## - "eventual": The controller will wait for all jobs to finish before applying the
  ##   change. This can lead to a longer time to apply the change but it will ensure
  ##   that you don't have any overprovisioning of runners.
  updateStrategy: "immediate"
```

Reply:
This looks great! :+1:

Reply:
Fixed